### PR TITLE
Add local branch sorting menu

### PR DESCRIPTION
**Pull request by hosaka**
- **PR Description**

Previously discussed in #3171 with @stefanhaller. I took on the advice and decided to not make the `sortMenuOption` a public type, passing a string array to the `CreateSortOrderMenu` achieves what we need with less code. If an option requested by the caller is not available, it will be excluded from the displayed menu.

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go generate ./...`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [ ] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

File: pkg/commands/git_commands/branch_loader.go
Comment (stefanhaller) [inline pkg/commands/git_commands/branch_loader.go:205]: When sorting by recency, branches that don't have a reflog entry for some reason used to have an empty recency column. Now they show the committer date, which is confusing. We should only set the recency here when not sorting by "recency"; in my approach I solved this by adding a `storeCommitterDateAsRecency` parameter to `obtainBranch`, see https://github.com/jesseduffield/lazygit/commit/268c56bd3e209141b0563747a2b26e419892218e#diff-6daca80788b73b1a2870465493179bde13e0be1ef2612f377f8dddb38dd6986eR193. (Would need another change to the branch_loader tests.)

File: pkg/gui/controllers/helpers/refs_helper.go
Comment (stefanhaller) [inline pkg/gui/controllers/helpers/refs_helper.go:139]: Since it's a programming error to pass a string that doesn't occur in `availableSortOptions`, I'd add an `else` with a panic here (which seems to be go's closest equivalent to an assert in other languages).

File: pkg/integration/tests/branch/sort_local_branches.go
Comment (stefanhaller) [inline pkg/integration/tests/branch/sort_local_branches.go:46]: Would be nice to also switch to alphabetical. This requires assigning the committer dates differently so that the order is actually different.

File: pkg/commands/git_commands/branch_loader_test.go
Comment (stefanhaller) [inline pkg/commands/git_commands/branch_loader_test.go:21]: We are mocking what git returns to us here, and an empty string is very unrealistic, git will never return that. I'd suggest to return a real number that could happen in practice; one way to do that would be
```
	now := time.Now().Unix()
	twoAndAHalfHoursAgo := strconv.Itoa(int(now - 2.5*60*60))
```
This will result in a recency string of `2h`.

If you follow the suggestion to add a `storeCommitterDateAsRecency` parameter, it would be good to add that to the scenario struct and add one more test case where it is true.

**Review by stefanhaller — COMMENTED** [review]
Nice! A few things below.

**Comment by hosaka** [comment]
Thanks for the suggestions! I'll try to go through them before the holidays kick in :)

File: pkg/commands/git_commands/branch_loader.go
Comment (hosaka) [reply pkg/commands/git_commands/branch_loader.go]: Totally missed that, fixed in https://github.com/jesseduffield/lazygit/pull/3182/commits/bea96a5babac6ae2a941e99a01446d7c23fc2c8e

File: pkg/gui/controllers/helpers/refs_helper.go
Comment (hosaka) [reply pkg/gui/controllers/helpers/refs_helper.go:139]: Fixed in https://github.com/jesseduffield/lazygit/commit/bea96a5babac6ae2a941e99a01446d7c23fc2c8e#diff-5ce8d32b6a73dd700e90181f144d5142ec388b1142ae13db669773b8c928dd95R137

File: pkg/commands/git_commands/branch_loader_test.go
Comment (hosaka) [reply pkg/commands/git_commands/branch_loader_test.go]: Fixed in https://github.com/jesseduffield/lazygit/commit/bea96a5babac6ae2a941e99a01446d7c23fc2c8e#diff-db04dda60e2b65b04e8136c601b4549909b8f0bd6c7dbc852fb436374b24ab6fR91

File: pkg/integration/tests/branch/sort_local_branches.go
Comment (hosaka) [reply pkg/integration/tests/branch/sort_local_branches.go:67]: Fixed in https://github.com/jesseduffield/lazygit/pull/3182/commits/de3d48f98f7121be1f073c6ba656a3b6bf85d60f. I've added a new "master" branch that stays selected at the top, instead of modifying the commit dates.

**Comment by hosaka** [comment]
Addressed the suggestions in the fixups.

File: pkg/commands/git_commands/branch_loader.go
Comment (stefanhaller) [inline pkg/commands/git_commands/branch_loader.go]: I'd prefer if this stays here; it's good to see how the fields are unpacked in one place. This requires adding a time stamp to the mock data of all tests, which is preferable too because it's closer to what git returns in production. (Changed in 5f272433fabd)

File: pkg/commands/git_commands/branch_loader_test.go
Comment (stefanhaller) [inline pkg/commands/git_commands/branch_loader_test.go]: This should be called `storeCommitDateAsRecency`, it's one of the inputs to the function we're testing. (Changed in 5f272433fabd)

File: pkg/integration/tests/branch/sort_local_branches.go
Comment (stefanhaller) [inline pkg/integration/tests/branch/sort_local_branches.go]: No need to create the master branch, it's checked out by default. (Fixed in c5221ac915a5)

**Review by stefanhaller — APPROVED** [review]
Great, looks good now. Two more very small things; in the interest of saving another roundtrip I pushed fixups for them, see below. Hope you are ok with that.

Will merge if tests are green. Thanks again!

File: pkg/commands/git_commands/branch_loader.go
Comment (hosaka) [reply pkg/commands/git_commands/branch_loader.go]: Fair enough, if there is no problem adding the 2h timestamp to other scenarios in `branch_loader_test.go`.

**Comment by hosaka** [comment]
All good, thanks for pushing the updates! I might be taking a stab at https://github.com/jesseduffield/lazygit/issues/3166 next.

**Comment by stefanhaller** [comment]
That would be great. Let me know if you need any input, especially for the migration question (I think the rest should be straight-forward).

**Pull request merged**
